src/org/labkey/test/tests/core/admin/AllowedFileExtensionAdminTest.java

```diff
@@ -0,0 +1,618 @@
+package org.labkey.test.tests.core.admin;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.remoteapi.CommandException;
+
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.TestFileUtils;
+import org.labkey.test.categories.Git;
+import org.labkey.test.components.ext4.Window;
+import org.labkey.test.components.html.FileInput;
+import org.labkey.test.components.html.Input;
+import org.labkey.test.pages.core.admin.AllowedFileExtensionAdminPage;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.params.experiment.SampleTypeDefinition;
+import org.labkey.test.util.DataRegionTable;
+import org.labkey.test.util.Ext4Helper;
+import org.labkey.test.util.PortalHelper;
+import org.labkey.test.util.SampleTypeHelper;
+import org.labkey.test.util.exp.SampleTypeAPIHelper;
+import org.labkey.test.pages.announcements.InsertPage;
+import org.openqa.selenium.Alert;
+import org.openqa.selenium.WebElement;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.labkey.test.util.DataRegionTable.DataRegion;
+
+@Category({Git.class})
+public class AllowedFileExtensionAdminTest extends BaseWebDriverTest
+{
+    private final File CSV_FILE = TestFileUtils.getSampleData("fileTypes/csv_sample.csv");
+    private final File TSV_FILE = TestFileUtils.getSampleData("fileTypes/tsv_sample.tsv");
+    private final File TXT_FILE = TestFileUtils.getSampleData("fileTypes/sample.txt");
+    private final File XLS_FILE = TestFileUtils.getSampleData("fileTypes/xls_sample.xls");
+    private final File XLSX_FILE = TestFileUtils.getSampleData("fileTypes/xlsx_sample.xlsx");
+
+    private final Map<String, File> fileMap = Map.of(
+            ".csv", CSV_FILE,
+            ".tsv", TSV_FILE,
+            ".txt", TXT_FILE,
+            ".xls", XLS_FILE,
+            ".xlsx", XLSX_FILE
+    );
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        AllowedFileExtensionAdminTest init = getCurrentTest();
+        init.doSetup();
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest)
+    {
+        _containerHelper.deleteProject(getProjectName(), afterTest);
+        try
+        {
+            deleteAllAllowedFileExtension();
+        }
+        catch (IOException | CommandException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void doSetup()
+    {
+        _containerHelper.createProject(getProjectName(), null);
+
+        goToProjectHome();
+        new PortalHelper(getDriver()).addWebPart("Files");
+        new PortalHelper(getDriver()).addWebPart("Sample Types");
+        new PortalHelper(getDriver()).addWebPart("Lists");
+        new PortalHelper(getDriver()).addWebPart("Messages List");
```

Comment:
Each `addWebPart` enters and exits page admin mode. Wrap in `doInAdminMode` to avoid all the extra navigations.
```suggestion
        new PortalHelper(driver).doInAdminMode(ph -> {
            ph.addWebPart("Files");
            ph.addWebPart("Sample Types");
            ph.addWebPart("Lists");
            ph.addWebPart("Messages List");
        });
```

Reply:
Done.